lib/types.nix

```diff
@@ -336,10 +336,16 @@ rec {
         in if isDerivation res then res else toDerivation res;
     };
 
-    shellPackage = package // {
-      check = x: (package.check x) && (hasAttr "shellPath" x);
+    packageWithAttr = type: attr: package // {
+      name = "${type}Package";
+      description = "${type} package";
+      check = x: package.check x && hasAttr attr x;
```

Comment:
Can we generate a meaningful error message here? I know that this will complain about the invalid type but as it is likely that someone adds a new editor without knowing about the new passthru's it might be good to tell the user what is required.

Reply:
I think the most meaningful error message we're likely to end up with is "package with the passthru ${attr}", and I'm unsure as to whether that's any more clear than the error that's already produced.

Reply:
I was thinking about something like:

```
The package you provided as defaults.editor does not have the required attribute `editorCommand`. Please select the correct package or add the passthru to the package as documented in the option.
```

NIx is often already bad enough user experience for new users so it is probably worth being helpful in error messages.